src/components/Switch/Switch.tsx

```diff
@@ -10,89 +11,85 @@ export type ToggleProps = SwitchProps & {
 }: ToggleProps) => {
-    const { Icon, View } = useMolecules();
+    const { Icon } = useMolecules();
     const { actionsRef, focused, hovered, pressed } = useActionState();
 
-    const componentStyles = useComponentStyles('Switch', style, {
-        states: {
-            selected_hovered: !!value && !!hovered,
-            selected_disabled: !!value && !!disabled,
-            selected_pressed: !!value && !!pressed,
-            selected_focused: !!value && !!focused,
-            selected: !!value,
-            disabled: !!disabled,
-            hovered: !!hovered,
-            focused: !!focused,
-            pressed: !!pressed,
-        },
+    const [value, onChange] = useControlledValue({
+        value: valueProp,
+        onChange: onValueChange as any,
+        defaultValue: false,
+        disabled,
     });
+
+    const componentStyles = useComponentStyles(
+        'Switch',
+        [
+            style,
+            trackColor?.true && { checkedColor: trackColor?.true },
+            trackColor?.false && { checkedColor: trackColor?.false },
+            thumbColor && { thumbColor },
+        ],
+        {
+            states: {
+                selected_hovered_pressed: !!value && !!hovered && !!pressed,
+                selected_focused_pressed: !!value && !!focused && !!pressed,
+                selected_hovered: !!value && !!hovered,
+                selected_disabled: !!value && !!disabled,
+                selected_pressed: !!value && !!pressed,
+                selected_focused: !!value && !!focused,
+                selected: !!value,
+                hovered_pressed: !!hovered && !!pressed,
+                disabled: !!disabled,
+                hovered: !!hovered,
+                focused: !!focused,
+                pressed: !!pressed,
+            },
+        },
+    );
     const toggleMarginAnimation = useRef(new Animated.Value(value ? 0 : 1)).current;
     const toggleSizeAnimation = useRef(new Animated.Value(value ? 0 : 1)).current;
 
-    const moveToggle = toggleMarginAnimation.interpolate({
+    const thumbPosition = toggleMarginAnimation.interpolate({
         inputRange: [0, 1],
         outputRange: [
-            !value && pressed && !unCheckedIcon
-                ? 0
-                : !value && unCheckedIcon
-                ? size * 0.04
-                : size * 0.2,
+            !value && pressed && !unCheckedIcon ? 0 : unCheckedIcon ? size * 0.04 : size * 0.2,
             pressed ? size * 0.74 : size * 0.81,
         ],
     });
 
-    const toggleSize = toggleSizeAnimation.interpolate({
+    const thumbSize = toggleSizeAnimation.interpolate({
         inputRange: [0, 1],
         outputRange: [
             !value && !pressed && !unCheckedIcon
                 ? size * 0.5
                 : value || unCheckedIcon
                 ? size * 0.74
-                : size * 0.84,
+                : size * 0.88,
             pressed && !value ? size : pressed && value ? size * 0.81 : size * 0.64,
```

Comment:
Right now, these numbers don't mean anything to me. They seem arbitrary. 
It would be easier to read if these are properly named consts.